src/main/java/com/melihsurkmez/secondhand/ecommerce/service/UserService.java

```diff
@@ -0,0 +1,67 @@
+package com.melihsurkmez.secondhand.ecommerce.service;
+
+import com.melihsurkmez.secondhand.ecommerce.DTO.CreateUserRequest;
+import com.melihsurkmez.secondhand.ecommerce.DTO.UpdateUserRequest;
+import com.melihsurkmez.secondhand.ecommerce.DTO.UserResponse;
+import com.melihsurkmez.secondhand.ecommerce.DTO.UserResponseConverter;
+import com.melihsurkmez.secondhand.ecommerce.exception.UserNotFound;
+import com.melihsurkmez.secondhand.ecommerce.model.User;
+import com.melihsurkmez.secondhand.ecommerce.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final UserResponseConverter userResponseConverter;
+
+
+    public UserService(UserRepository userRepository, UserResponseConverter userResponseConverter) {
+        this.userRepository = userRepository;
+        this.userResponseConverter = userResponseConverter;
+    }
+
+    private User findUserByEmail(String email){
+
+        return userRepository.findBymail(email).orElseThrow(() -> new UserNotFound("User with email "+email+" not found!"));
+    }
+
+    public UserResponse getUserbyEmail(String email) {
+
+        User user= findUserByEmail(email);
+        return userResponseConverter.convert(user);
+
+    }
+
+    public UserResponse createUser(CreateUserRequest newUser) {
+
+        User user= new User(newUser.getFirstName(),newUser.getMiddleName(),newUser.getLastName(),newUser.getEmail());
+
+        return userResponseConverter.convert(userRepository.save(user));
+
+    }
+
+    public UserResponse updateUser(String email,UpdateUserRequest updatedUser) {
+
+        User user = findUserByEmail(email);
+        User updatedNewUser=new User(user.getId(),updatedUser.getFirstName(),updatedUser.getMiddleName(),updatedUser.getLastName(),email);
```

Comment:
Why do you need to create a new `User updatedNewUser` object? Can't you use the object that you find with `findUserByEmail`? In fact, write a mapper for this purpose but be careful and do not forget to check null or empty variables.

```.jaav
User user = findUserByEmail(email);
user.setFirstName(updatedUser.getFirstName());
user.setMiddleName(updatedUser.getMiddleName());
...
```

Reply:
![image](https://user-images.githubusercontent.com/100957608/215030448-7a775481-0a5e-46f0-8bd9-08e0896e9efc.png)

I assigned the updated user to the user and saved it again.